src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,199 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Handles parsing of user input into commands, tasks, and task details for the Duke application.
+ * Provides utility methods to interpret user commands and to extract relevant details for task creation.
+ */
+public class Parser {
+
+    /**
+     * Parses the command from user input.
+     *
+     * @param userInput The user input string.
+     * @return The CommandType corresponding to the parsed command.
+     * @throws DukeException If the command is not recognized.
+     */
+    public static CommandType parseCommand(String userInput) throws DukeException {
+        String[] parts = userInput.trim().split("\\s+", 2);
+        String commandStr = parts[0].toUpperCase();
+
+        try {
+            return CommandType.valueOf(commandStr);
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Processes the user input and creates the appropriate Command object.
+     *
+     * @param userInput The user input string.
+     * @return The Command object corresponding to the user input.
+     * @throws DukeException If there is an issue processing the command.
+     */
+    public static Command processCommand(String userInput) throws DukeException {
+
+        try {
+            CommandType command = Parser.parseCommand(userInput);
+
+            switch (command) {
+            case LIST:
+                return new ListCommand();
+            case MARK:
+                return new MarkCommand();
+            case UNMARK:
+                return new UnmarkCommand();
+            case DELETE:
+                return new DeleteCommand();
+            case BYE:
+                return new ExitCommand();
+            case TODO:
+                // Fallthrough
+            case DEADLINE:
+                // Fallthrough
+            case EVENT:
+                return new AddCommand();
+            case FIND:
+                return new FindCommand();
+            default:
+                throw new DukeException("I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
```

Comment:
I admire your implementation of the switch statement in this section of your code to determine whether the user has entered the correct input. Personally felt that it was a clever approach that I had not considered before.